src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -48,6 +48,12 @@ public Task RunAsync(ActionRunStage stage)
             // Add each composite action step to the front of the queue
             int location = 0;
```

Comment:
do you need `location` anymore?
also, you don't need `tempDirectory`

Reply:
good point - thanks!